src/Controller/DeveloperAppListBuilderForDeveloper.php

```diff
@@ -96,7 +97,7 @@ protected function getEntityIds(array $headers = [], UserInterface $user = NULL)
     // either there is connection error or the site is out of sync with
     // Apigee Edge.
     if ($developerId === NULL) {
-      return [];
+      throw new DeveloperDoesNotExistException("Edge user not found.");
```

Comment:
* Single quote
* Introduce a new developer email parameter to the constructor of this exception.
* Default message returned by the exception (if no $message is provided as the second parameter) should be something like: `Developer with %s email address not found.` Therefore you do not need to the repeat the message text every time when this exceptions needs to be thrown.
* If message override provided in the construction then $developerEmail should be available as a token. Ex.: throw new DeveloperDoesNotExistException('foobar@example.com', 'This prints out the email: %s') -> "This prints out the email: foobar@example.com"
* It should extend the ApiException, because it is actually an API issue and if it extends you do not need to check if the exception is instance of `DeveloperDoesNotExistException` in the route subscriber.